src/main/java/com/ebay/myriad/scheduler/TaskFactory.java

```diff
@@ -57,6 +58,30 @@ public NMTaskFactoryImpl(MyriadConfiguration cfg, TaskUtils taskUtils) {
             this.taskUtils = taskUtils;
         }
 
+        //Utility function to get the first three ports of an offer
+        private static NMPorts getPorts(Offer offer) {
+            HashSet<Long> ports = new HashSet<Long>();
+            for (Resource resource : offer.getResourcesList()){
+                if (resource.getName().equals("ports")){
+                    Iterator<Value.Range> itr = resource.getRanges().getRangeList().iterator();
```

Comment:
`getRanges.getRangeList()` returns a list of objects of type `Range`.  It has two methods `getBegin()` and `getEnd()`.  There could be exactly one `Range` in `getRanges.getRangeList` that would have 1000 ports.  This would cause the `if` statement to fail likewise `subList` wouldn't work as expected as the would be a sublist of `Range` not `Long`.  It may warrant a comment?

In general I expect to get one range, loop through the first 4 longs between `beginRange` and `endRange` and then be out.  But its possible due to other frameworks etc, that we get 500 ranges of size 1, in which case we loop through the the first 4 ranges.  Either way its bounded by 4.